source/backend/database/models/completedRecipesModel.js

```diff
@@ -13,4 +13,31 @@ async function getCompletedRecipes(userId){
   return completedRecipes;
 }
 
-module.exports = { getCompletedRecipes };
\ No newline at end of file
+/**
+ *  Adds a new recipe to the user's completed list by recipeID
+ * @param recipeId
+ * @returns {Promise<awaited Knex.QueryBuilder<TRecord, TResult>>}
+ */
+ async function addNewRecipe(recipeId, userId){
+  
+  await db.transaction(async (transaction) => {
+    try {
+      await db('completedRecipes')
+            .transacting(transaction)
+            .where({userId, recipeId})
+            .insert();
+      await transaction.commit();
+    } catch (err) {
+      console.log(err);
+      await transaction.rollback();
+    }
+  });

```

Comment:
Do you need a transaction if you are adding one recipe to one table? Since we are adding an existing recipe from recipes table to completedRecipes, I don't think we need to do a transaction?

Reply:
+1

Reply:
Also to make the function more clear it can be renamed to addNewCompletedRecipe or addCompletedRecipe().

Reply:
The standard naming convention we are following omits the first class citizen, since the file name already gives that information. Therefore, for simplicity, this function can just be called `insert`

Reply:
^ 👍